src/index.d.ts

```diff
@@ -0,0 +1,158 @@
+declare module 'react-form' {
+  import {
+    ChangeEventHandler,
+    ComponentType,
+    Dispatch,
+    FocusEventHandler,
+    FormEventHandler,
+    HTMLProps,
+    Provider,
+    SetStateAction,
+  } from 'react'
+
+  type Debounce = <T>(fn: (...args: any[]) => T, wait: number) => Promise<T>
+  type ValidatorReturn = string | false | undefined
+  type OptionalPromise<T> = Promise<T> | T
+
+  interface FormMeta {
+    error: string | any
+    isSubmitting: boolean
+    isDirty: boolean
+    isSubmitted: boolean
+    submissionAttempts: number
+    isValid: boolean
+    fieldsAreValidating: boolean
+    fieldsAreValid: boolean
+    canSubmit: boolean
+    [k: string]: any
+  }
+
+  type SetFormMeta = Partial<FormMeta> | ((previousMeta: FormMeta) => FormMeta)
+
+  interface FormInstance<T> {
+    Form: ComponentType<Omit<HTMLProps<HTMLFormElement>, 'onSubmit'>>
+    values: T
+    meta: FormMeta
+    formContext: FormInstance<T>
+    reset: () => void
+    setMeta: (value: SetFormMeta) => void
+    handleSubmit: FormEventHandler<HTMLFormElement>
+    debounce: Debounce
+    setValues: Dispatch<SetStateAction<T>>
+    runValidation: () => void
+    getFieldValue: <V = any>(fieldPath: keyof T | string) => V
```

Comment:
Whilst this allows the _path of least resistance_, I would suggest considering:

```suggestion
    getFieldValue: <V = unknown>(fieldPath: keyof T) => V
```

This forces the _consumer_ to do dynamic type checks on the field value. I also suggest _not_ allowing a _random string_, because that doesn't work with type broadning. `'a' | 'b' | string` will always allow _any_ string, even if `T` is set correctly.

Reply:
If we did this, we'd be saying that you can't access nested values when giving the function a string path to access a sub-value (in the case that the forms values are a series of nested objects and/or arrays). This kind of pattern ( implemented here: https://github.com/tannerlinsley/react-form/blob/master/src/utils.js#L35) is fundamentally incompatible with TypeScript, so that's why I left the `string` in there. But, that also makes the types for this function not really useful.

So, stricter type checking and skip that feature, or looser but support it? I almost lean towards the former. @tannerlinsley, do you have any thoughts?

Reply:
I'm personally *not* a fan of the path.access.thing, because it's so easy to do wrongly, and you won't get type safety anywhere. There is no way to do it. 

That said, if the looser variant is chosen, I still recommend `unknown`.